src/catsDb.js

```diff
@@ -77,14 +81,14 @@ class CatRepository {
         if (catIndex >= 0) {
             return catIndex; 
         } else {
-            return null; 
+            return null; // TODO - check if we can DRY this check for id. We are passing a lot of 'nulls' around. catIndex is either an index or -1. 
```

Comment:
I'd go further and say that you could alter all the places that currently check null to check if the index is -1 instead. By convention if an index is -1 in programming it means that the element is not present in an array.

Reply:
Done.